--watch is broken in release-2.0.5

1. create file `app.ts`
2. start compiler in watch mode, i.e. `node built/local/tsc.js --watch app.ts`
3. save the file

**Expected:** no errors, output file is written
**Actual:** error `error TS5033: Could not write file 'app.js': ENOENT: no such file or directory, mkdir ''` occured
`

--- a/src/compiler/sys.ts
+++ b/src/compiler/sys.ts
@@ -611,7 +611,7 @@ namespace ts {
             const originalWriteFile = sys.writeFile;
             sys.writeFile = function(path, data, writeBom) {
                 const directoryPath = getDirectoryPath(normalizeSlashes(path));
-                if (!sys.directoryExists(directoryPath)) {
+                if (directoryPath && !sys.directoryExists(directoryPath)) {
                     recursiveCreateDirectory(directoryPath, sys);
                 }
                 originalWriteFile.call(sys, path, data, writeBom);

